### PR TITLE
Return Auth tokens on user registration

### PR DESCRIPTION
**Pull request by whyderrick**
This commit makes a change to the RegisterUser migration, allowing it to return an access_token and refresh_token from Doorkeeper. It uses a wrapper service `LocalAuthClient` to return the `DoorkeeperApplication` we're using for local registrations.

A consumer of the API triggering this migration can retrieve user data, but is returned an access_token and refresh_token.

It's worth noting that, at this point, this app isn't following the full OAuth flow. We could, but we don't expect to be an external provider anytime soon, so we're effectively using Doorkeeper to create token, but skipping device registration and access grant flows.

### CODE REVIEW

File: app/services/local_auth_client.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  def self.create_token(user)
+    new(user).create_token
+  end
+
+  def initialize(user)
+    @user_id = user.id
+  end
+
+  def create_token
+    Doorkeeper::AccessToken.create(
```
Comment (composerinteralia) [inline app/services/local_auth_client.rb]: Is `Doorkeep::AccessToken` an ActiveRecord model? If so, is it OK if this silently fails to persist, or would it be better to raise an error with `create!`?

File: app/services/local_auth_client.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class LocalAuthClient
+  NAME = "SparkJoy Local Client"
+  REDIRECT_URI = "https://sparkjoy.com/oauth"
```
Comment (composerinteralia) [inline app/services/local_auth_client.rb]: Is this the same in all environments?

File: config/environments/development.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
   config.cache_classes = false
   config.eager_load = false
   config.consider_all_requests_local = true
+  config.action_controller.allow_forgery_protection = false
```
Comment (composerinteralia) [inline config/environments/development.rb]: Why was this necessary?

File: spec/graphql/mutations/register_user_spec.rb [hunk 4]
```diff
@@ -20,34 +19,57 @@
           variables: { email: user.email, password: user.password },
         )
 
+        expect(before_count).to eq 0
```
Comment (composerinteralia) [inline spec/graphql/mutations/register_user_spec.rb]: The RSpec [`change` matcher](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher) might be handy here:

```rb
expect { some_code }.to change(User, :count).by(1)
```

File: config/environments/development.rb (hunk 3 above)
Comment (whyderrick) [reply config/environments/development.rb]: It's not anymore. It was causing csrf failures when I was initially using the OAuth redirects. 

It's a change I'll need eventually since this is an api that I initialized without the `api_mode` flag (I'll need an Admin dashboard, but I learned later that I can add back the middleware I need when I add in Administrate). 

I'm planning a follow-on PR where I properly configure this app as an API and add the requisite middleware when add in Administrate.

File: app/services/local_auth_client.rb (hunk 2 above)
Comment (whyderrick) [reply app/services/local_auth_client.rb]: It's not, but it's also effectively a dummy variable we have to give Doorkeeper to create a client. I'll be breaking environment variables into separate files when I integrate external providers.

File: app/services/local_auth_client.rb (hunk 1 above)
Comment (whyderrick) [reply app/services/local_auth_client.rb]: It is a model. Good point.

Comment (whyderrick) [reply app/services/local_auth_client.rb]: A little more nuance. For this use, in the service, it makes sense to fail loudly. In the mutation, since we're using this pattern for [errors as data](https://graphql-ruby.org/mutations/mutation_errors#errors-as-data), we don't want to fail loudly, and instead return the errors to the client.
